domain/secretbackend/state/state.go

```diff
@@ -338,17 +363,58 @@ ORDER BY b.name`, SecretBackendRow{})
 	if err != nil {
 		return nil, fmt.Errorf("cannot list secret backends: %w", err)
 	}
-	backends := rows.toSecretBackends()
+	return rows.toSecretBackends(), errors.Trace(err)
```

Comment:
We are already checking the error above. We either want to return nil here or remove the error check on 363